dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java

```diff
@@ -916,12 +912,10 @@ private String joinGlobalParams(String parentGlobalParams, String subGlobalParam
      */
     private void initTaskInstance(TaskInstance taskInstance) {
 
-        if (!taskInstance.isSubProcess()) {
-            if (taskInstance.getState().typeIsCancel() || taskInstance.getState().typeIsFailure()) {
-                taskInstance.setFlag(Flag.NO);
-                updateTaskInstance(taskInstance);
-                return;
-            }
+        if (!taskInstance.isSubProcess() && taskInstance.getState().typeIsCancel() || taskInstance.getState().typeIsFailure()) {
```

Comment:
this should be changed like this:
   if (!taskInstance.isSubProcess() && (taskInstance.getState().typeIsCancel() || taskInstance.getState().typeIsFailure())) {

how do you think?

Reply:
That's true. I missed it

Reply:
Done